delta-app/src/test/java/io/cdap/delta/app/DeltaPipelineDBStateStoreTest.java

```diff
@@ -0,0 +1,680 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.delta.app;
+
+import com.google.gson.Gson;
+import io.cdap.cdap.api.ServiceDiscoverer;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.common.http.DefaultHttpRequestConfig;
+import io.cdap.cdap.common.utils.Tasks;
+import io.cdap.cdap.proto.ProgramRunStatus;
+import io.cdap.cdap.proto.artifact.AppRequest;
+import io.cdap.cdap.proto.id.ApplicationId;
+import io.cdap.cdap.proto.id.NamespaceId;
+import io.cdap.cdap.test.ApplicationManager;
+import io.cdap.cdap.test.ServiceManager;
+import io.cdap.cdap.test.TestConfiguration;
+import io.cdap.cdap.test.WorkerManager;
+import io.cdap.common.http.HttpMethod;
+import io.cdap.common.http.HttpRequest;
+import io.cdap.common.http.HttpRequests;
+import io.cdap.common.http.HttpResponse;
+import io.cdap.delta.api.ChangeEvent;
+import io.cdap.delta.api.DDLEvent;
+import io.cdap.delta.api.DDLOperation;
+import io.cdap.delta.api.DMLEvent;
+import io.cdap.delta.api.DMLOperation;
+import io.cdap.delta.api.DeltaPipelineId;
+import io.cdap.delta.api.Offset;
+import io.cdap.delta.api.SourceTable;
+import io.cdap.delta.app.service.AssessmentService;
+import io.cdap.delta.proto.DeltaConfig;
+import io.cdap.delta.proto.ParallelismConfig;
+import io.cdap.delta.proto.PipelineReplicationState;
+import io.cdap.delta.proto.PipelineState;
+import io.cdap.delta.proto.RetryConfig;
+import io.cdap.delta.proto.Stage;
+import io.cdap.delta.proto.StateStoreConfig;
+import io.cdap.delta.proto.TableReplicationState;
+import io.cdap.delta.proto.TableState;
+import io.cdap.delta.store.StateStoreFactory;
+import io.cdap.delta.test.DeltaPipelineTestBase;
+import io.cdap.delta.test.mock.FailureTarget;
+import io.cdap.delta.test.mock.FileEventConsumer;
+import io.cdap.delta.test.mock.MockErrorTarget;
+import io.cdap.delta.test.mock.MockSource;
+import io.cdap.delta.test.mock.MockTarget;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Tests for delta pipelines
+ */
+public class DeltaPipelineDBStateStoreTest extends DeltaPipelineTestBase {
```

Comment:
Seems like these are copies of the existing tests? Should refactor common code to a common place, not copy-paste.